06_puppy/sirigithub/stores_test.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type storesSuite struct {
+	suite.Suite
+	store     Storer
+	newStorer func() Storer
+	puppy1    Puppy
+}
+
+func (suite *storesSuite) SetupTest() {
+	suite.store = suite.newStorer()
+	suite.puppy1 = Puppy{Breed: "cavalier", Color: "White", Value: "$1700"}
+	suite.store.CreatePuppy(&suite.puppy1)
+}
+func (suite *storesSuite) TestCreatePuppySuccesful() {
+	//given
+	assert := assert.New(suite.T())
+	newPuppy := Puppy{Breed: "Dobberman", Color: "Black", Value: "$800"}
+	//when
+	id := suite.store.CreatePuppy(&newPuppy)
+	//then
+	assert.True(newPuppy.ID == suite.puppy1.ID+1, "Puppy Id increments sequentially")
+	puppy, err := suite.store.ReadPuppy(id)
+	if assert.NoError(err, "Read of created puppy should be successful") {
+		assert.Equal(puppy, &newPuppy)
+	}
+}
+func (suite *storesSuite) TestReadPuppySuccesful() {
+	//given
+	assert := assert.New(suite.T())
+	//when
+	puppy, err := suite.store.ReadPuppy(suite.puppy1.ID)
+	//then
+	if assert.NoError(err, "Read of puppy should be successful") {
+		assert.Equal(puppy, &suite.puppy1)
+	}
+}
+func (suite *storesSuite) TestReadPuppyDoesNotExist() {
+	assert := assert.New(suite.T())
+	_, err := suite.store.ReadPuppy(100)
+	assert.Error(err, "Read of non existing puppy Id should throw an error")
```

Comment:
As Julia mentioned in a previous comment, "throw" is the wrong word to use here as Go does not have "throw". "returns" would be a better verb.